source/common/secret/sds_api.h

```diff
@@ -72,11 +72,23 @@ typedef std::shared_ptr<SdsApi> SdsApiSharedPtr;
  */
 class TlsCertificateSdsApi : public SdsApi, public TlsCertificateConfigProvider {
 public:
+  static SdsApiSharedPtr create(const LocalInfo::LocalInfo& local_info,
+                                Event::Dispatcher& dispatcher, Runtime::RandomGenerator& random,
+                                Stats::Store& stats, Upstream::ClusterManager& cluster_manager,
+                                Init::Manager& init_manager,
+                                const envoy::api::v2::core::ConfigSource& sds_config,
+                                const std::string& sds_config_name,
```

Comment:
can we use provider_context in create() function so less number of argument to pass?

Reply:
Done.